sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -217,6 +219,20 @@ public DataSourceConfiguration withPassword(String password) {
       return builder().setPassword(password).build();
     }
 
+    /**
+     * Sets the connection properties passed to driver.connect(...).
+     * Format of the string must be [propertyName=property;]*
+     *
+     * <p>NOTE - The "user" and "password" properties can be add via {@link #withUsername(String)},
+     * {@link #withPassword(String)}, so they do not need to be included here.
+     * @param connectionProperties
+     * @return
+     */
+    public DataSourceConfiguration withConnectionProperties(String connectionProperties) {
+      checkArgument(connectionProperties != null, "connectionProperties is null");
```

Comment:
Can you use the same kind of message as for the other `with*` methods ? Something like `DataSourceConfiguration.create(driver, url).withConnectionProperties(connectionProperties) called with null connectionProperties)`.